clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -421,6 +446,18 @@ private AbstractIterator<FileChannelRecordBatch> batchIterator(int start) {
         return new RecordBatchIterator<>(inputStream);
     }
 
+    /**
+     * Try populating OS page cache with file content
+     */
+    public void prepareForRead() throws IOException {
+        if (DEVNULL_PATH != null) {
+            long size = Math.min(channel.size(), end) - start;
```

Comment:
Isn't `size` member in FileRecords precomputed for FileRecords during construction and maintained whenever there is more mutation? Hence, we can simply use `this.size` here. No?

Reply:
Good point. `this.size` should work here. Let me fix